lib/srv/alpnproxy/helpers_test.go

```diff
@@ -38,8 +34,8 @@ import (
 
 	"github.com/gravitational/teleport/api/types"
 	"github.com/gravitational/teleport/lib/auth"
-	"github.com/gravitational/teleport/lib/defaults"
 	"github.com/gravitational/teleport/lib/srv/alpnproxy/common"
+	"github.com/gravitational/teleport/lib/srv/alpnproxytest"
```

Comment:
This is not common patter to introduce the `*test` package.  Maybe instead creating a *testpackage we can create `helper` file and extract to some kind of test utils package. 
Also one of the "Go Proverbs" is 
> [a little copying is better than a little dependency"](https://www.efekarakus.com/2021/09/23/a-little-copying-is-better-than-a-little-dependency.html) 
and if the function is really small it better to copy then logic instead of introducing artificial dependency between package.

Reply:
I think I'll follow the proverb and instead of extracting alpnproxytest, I'll just copy relevant bits to gatewaytest. I think I overestimated the complexity of those helpers. Now that I understand the code around keys and certs better, it doesn't seem that outlandish to me to just call the relevant x509, tls, tlsca and pem functions directly in gateway tests.

---

Re `*test` package not being a common pattern, is the issue here that the functions in alpnproxytest are too generic to be in a package called alpnproxytest? They relate to certs and are not specific to alpnproxy after all.

I created gatewaytest in lib/teleterm based on net/http/httptest and I think it fulfills its role quite well – whenever I have tests that interact with `gateway` in some way, I can import `gatewaytest` to reuse some functions that are useful in tests that deal with gateways.